addons/web/static/src/views/fields/fields.scss

```diff
@@ -65,6 +65,10 @@
     &.o_input, .o_input, &.o_form_label {
         color: $danger;
```

Comment:
Not strictly in the scope of the PR , but since we are improving the overall invalid field contrast we should use the danger colors fine-tuned for text here 
https://github.com/odoo/odoo/blob/17.0/addons/web/static/src/scss/primary_variables.scss#L92